src/tools/compiletest/src/directives/handlers.rs

```diff
@@ -67,6 +75,338 @@ fn make_directive_handlers_map() -> HashMap<&'static str, Handler> {
+                if known_bug == "unknown"
+                    || known_bug.split(',').all(|issue_ref| {
+                        issue_ref
+                            .trim()
+                            .split_once('#')
+                            .filter(|(_, number)| {
+                                // (preserve line breaks)
+                                number.chars().all(|digit| digit.is_numeric())
+                            })
+                            .is_some()
+                    })
+                {
+                    props.known_bug = true;
+                } else {
+                    panic!(
+                        "Invalid known-bug value: {known_bug}\nIt requires comma-separated issue references (`#000` or `chalk#000`) or `known-bug: unknown`."
+                    );
+                }
+            } else if config.parse_name_directive(ln, KNOWN_BUG) {
+                panic!(
+                    "Invalid known-bug attribute, requires comma-separated issue references (`#000` or `chalk#000`) or `known-bug: unknown`."
+                );
+            }
+        }),
+        handler(TEST_MIR_PASS, |config, ln, props| {
+            config.set_name_value_directive(
+                // (preserve line breaks)
+                ln,
+                TEST_MIR_PASS,
+                &mut props.mir_unit_test,
+                |s| s.trim().to_string(),
+            );
+        }),
+        handler(REMAP_SRC_BASE, |config, ln, props| {
+            config.set_name_directive(ln, REMAP_SRC_BASE, &mut props.remap_src_base);
+        }),
+        handler(LLVM_COV_FLAGS, |config, ln, props| {
+            if let Some(flags) = config.parse_name_value_directive(ln, LLVM_COV_FLAGS) {
+                props.llvm_cov_flags.extend(split_flags(&flags));
+            }
+        }),
+        handler(FILECHECK_FLAGS, |config, ln, props| {
+            if let Some(flags) = config.parse_name_value_directive(ln, FILECHECK_FLAGS) {
+                props.filecheck_flags.extend(split_flags(&flags));
+            }
+        }),
+        handler(NO_AUTO_CHECK_CFG, |config, ln, props| {
+            config.set_name_directive(ln, NO_AUTO_CHECK_CFG, &mut props.no_auto_check_cfg);
+        }),
+        handler(ADD_MINICORE, |config, ln, props| {
+            props.update_add_minicore(ln, config);
+        }),
+        handler(MINICORE_COMPILE_FLAGS, |config, ln, props| {
+            if let Some(flags) =
+                // (preserve line breaks)
+                config.parse_name_value_directive(ln, MINICORE_COMPILE_FLAGS)
+            {
+                let flags = split_flags(&flags);
+                for flag in &flags {
+                    if flag == "--edition" || flag.starts_with("--edition=") {
+                        panic!("you must use `//@ edition` to configure the edition");
+                    }
+                }
+                props.minicore_compile_flags.extend(flags);
+            }
+        }),
```

Comment:
Suggestion: can you add a FIXME about unifying the normalization + denylist logic with `compile-flags`?